cmd/kwild/server/migration.go

```diff
@@ -60,14 +61,33 @@ func PrepareForMigration(ctx context.Context, kwildCfg *commonCfg.KwildConfig, g
 	logger.Info("Entering migration mode", log.String("migrate_from", kwildCfg.MigrationConfig.MigrateFrom))
 
 	snapshotFileName := config.GenesisStateFileName(kwildCfg.RootDir)
-	// check if genesis hash is set in the genesis config
-	if genesisCfg.DataAppHash != nil &&
-		genesisCfg.ConsensusParams.Migration.StartHeight != 0 &&
-		genesisCfg.ConsensusParams.Migration.EndHeight != 0 &&
-		validateGenesisState(snapshotFileName, genesisCfg.DataAppHash) {
-		// genesis state already downloaded. No need to poll for genesis state
+
+	// if the genesis state is already downloaded, then no need to poll for genesis state
+	_, err := os.Stat(snapshotFileName)
+	if err == nil {
 		logger.Info("Genesis state already downloaded", log.String("genesis snapshot", snapshotFileName))
+
+		if !validateGenesisState(snapshotFileName, genesisCfg.DataAppHash) {
```

Comment:
`validateGenesisState` could use an update.  I don't think it should itself call `failBuild` (it's not in build.go and `PrepareForMigration` can return errors without having to panic).  Also, a bunch of logic there is obsoleted now that we're ensuring it's existence here.  The empty apphash logic in `validateGenesisState` is also a little questionable given the current changes, plus the `bytes.Equal` in the final return covers the length an nilness checks implicitly.

Reply:
Oh yeah. Im just going to change this whole function since it is a bit convoluted in its usage